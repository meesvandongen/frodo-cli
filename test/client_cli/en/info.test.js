import cp from 'child_process';
import { promisify } from 'util';

const exec = promisify(cp.exec);
const CMD = 'frodo info --help';
const { stdout } = await exec(CMD);

test("CLI help interface for 'info' should be expected english", async () => {
  expect(stdout).toMatchSnapshot();
});
