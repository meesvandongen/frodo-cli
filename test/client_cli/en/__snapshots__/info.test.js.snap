// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI help interface for 'info' should be expected english 1`] = `
"Usage: frodo info [options] [host] [username] [password]

Print versions and tokens.

Arguments:
  host                  Access Management base URL, e.g.:
                        https://cdk.iam.example.com/am. To use a connection
                        profile, just specify a unique substring.
  username              Username to login with. Must be an admin user with
                        appropriate rights to manage authentication
                        journeys/trees.
  password              Password.

Options:
  --curlirize           Output all network calls in curl format.
  --debug               Debug output during command execution. If specified,
                        may or may not produce additional output helpful for
                        troubleshooting.
  -h, --help            Help
  -k, --insecure        Allow insecure connections when using SSL/TLS. Has no
                        effect when using a network proxy for https
                        (HTTPS_PROXY=http://<host>:<port>), in that case the
                        proxy must provide this capability. (default: Don't
                        allow insecure connections)
  -m, --type <type>     Override auto-detected deployment type. Valid values
                        for type:
                        classic:  A classic Access Management-only deployment
                        with custom layout and configuration.
                        cloud:    A ForgeRock Identity Cloud environment.
                        forgeops: A ForgeOps CDK or CDM deployment.
                        The detected or provided deployment type controls
                        certain behavior like obtaining an Identity Management
                        admin token or not and whether to export/import
                        referenced email templates or how to walk through the
                        tenant admin login flow of Identity Cloud and handle
                        MFA (choices: "classic", "cloud", "forgeops")
  -s, --scriptFriendly  Send output of operation to STDOUT in a script-friendly
                        format (JSON) which can be piped to other commands.
                        User messages/warnings are output to STDERR, and are
                        not piped. For example, to only get bearer token:
                        <<< frodo info my-tenant -s 2>/dev/null | jq -r
                        .bearerToken >>> (default: Output as plain text)
  --sa-id <sa-id>       Service account id.
  --sa-jwk-file <file>  File containing the java web key (jwk) associated with
                        the the service account.
  --verbose             Verbose output during command execution. If specified,
                        may or may not produce additional output.
"
`;
